include/circt/Dialect/SV/SVInOutOps.td

```diff
@@ -137,3 +137,20 @@ def ArrayIndexInOutOp
   let assemblyFormat =
     "$input`[`$index`]` attr-dict `:` type($input) `,` type($index)";
 }
+
+def PartSelectOp : SVOp<"part_select", [NoSideEffect]> {
+  let summary = "Address several contiguous bits of an inout type (e.g. a wire"
+                " or inout port). This is an indexed part-select operator."
+                "The base is an integer expression and the width is an "
+                " integer constant. The bits start from base and the number "
+                "of bits selected is equal to width. See SV Spec 11.5.1.";
+  let arguments = (ins InOutType:$input, HWIntegerType:$base, I32Attr:$width);
+  let results = (outs InOutType:$result);
+
+  let builders = [
+    OpBuilder<(ins "Value":$input, "Value":$base, "int32_t":$width)>
+  ];
+
```

Comment:
Is it necessary to verify the width of the input is a multiple of the slice width?  Does verilog care?

Reply:
Checked the spec and with `iverilog`, It seems there is no such constraint on the relation of `width` and `input`.  So, width can be anything.